torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -65,12 +65,24 @@ PyRRef::PyRRef(c10::intrusive_ptr<RRef> rref) : rref_(std::move(rref)) {
 
 PyRRef::PyRRef(const py::object& value)
     : PyRRef([&value]() {
+        jit::InferredType type_inferred = jit::tryToInferType(value);
+        TypePtr elem_type = nullptr;
+        if (type_inferred.success()) {
+          // If we could infer the type from the pyobject, we create
+          // the RRef with the IValue of that type.
+          elem_type = type_inferred.type();
```

Comment:
Is there a way to test this? Say I do `RRef(1)`, then the `RRef` should beholding an IValue type int instead of `py::object`.

Reply:
I did test this through the module interface, a local created RRef (without passing a torchscript function or sth) is now associated with the JIT type instead of a blind `PyObjectType`, so it's compatible with the module interface type we defined in the type comment. we can also pass a tensor to it and call some tensor methods afterward, I will add more tests cases once https://github.com/pytorch/pytorch/pull/32959 landed.